### ISSUE TITLE
Is fields macro useless?

### ISSUE THREAD

**Issue by paomian** [korma/Korma#340]
``` clojure
(defentity user
    (pk :id)
    (entity-fields :a :b :c))
```

``` clojure
(sql-only (select user (fields :a)))
;;=> "SELECT `user`.`a`, `user`.`b`, `user`.`c` FROM `user`"
```

**Comment by immoh** [comment]
It's not, it adds fields on top of fields defined in the entity:

```clj
(sql-only (select user (fields :a)))
;;=> "SELECT \"user\".\"a\", \"user\".\"b\", \"user\".\"c\", \"user\".\"a\" FROM \"user\""
```

**Comment by paomian** [comment]
I means how i change the default field which be select.

```clojure
(select user (fields :a))
```
just `select user.a from user` not add default field such as `b` `c`.

**Comment by SeriousBug** [comment]
@paomian My solution to the same problem was creating a function like this:

```clojure
(defn limit-fields
  [ent & fields]
  (assoc ent :fields fields))
```

Then, you can use this in your queries to override the fields.

```clojure
(select user
  (limit-fields :a)) ; Will now only select user.a
```

**Comment by immoh** [comment]
Or just don't define them as `entity-fields` if you don't want to always include them. This has been also discussed in #251.

**Comment by jdpopkin** [comment]
I started using Korma earlier today and I ran into the same misunderstanding. Having read the other issue, I like the fact that Korma's query manipulation functions are additive-only, but I think some extra documentation might be helpful. Specifically, I'd like it if the documentation for select queries mentioned that fields specified in `fields` are selected _in addition to_ those specified in `entity-fields`.
